File: solutions/zig/raindrops/1/raindrops.zig
const std = @import("std");

pub fn convert(buffer: []u8, n: u32) []const u8 {
    var fixed = std.io.fixedBufferStream(buffer);
    var out = fixed.writer();
    if (n % 3 == 0) {
        out.writeAll("Pling") catch unreachable;
    }
    if (n % 5 == 0) {
        out.writeAll("Plang") catch unreachable;
    }
    if (n % 7 == 0) {
        out.writeAll("Plong") catch unreachable;
    }
    if (fixed.pos == 0) {
        out.print("{}", .{n}) catch unreachable;
    }
    return fixed.getWritten();
}

File: solutions/clojure/raindrops/1/src/raindrops.clj
(ns raindrops)

(defn convert [input] ;; <- arglist goes here
      ;; your code goes here
  (let [factor3? (if (zero? (mod input 3)) "Pling" nil)
        factor5? (if (zero? (mod input 5)) "Plang" nil)
        factor7? (if (zero? (mod input 7)) "Plong" nil)]
    (if-not (or factor3? factor5? factor7?)
      (str input)
      (reduce (fn [acc elem] (if elem (str acc elem) acc)) "" [factor3? factor5? factor7?]))))
